packages/financial-templates-lib/src/helpers/GasEstimator.js

```diff
@@ -66,21 +77,31 @@ class GasEstimator {
         message: "Gas estimator updated",
         networkId: this.networkId,
         lastUpdateTimestamp: this.lastUpdateTimestamp,
-        currentFastPriceGwei: this.lastFastPriceGwei,
+        currentMaxFeePerGas: this.latestMaxFeePerGasGwei,
+        currentMaxPriorityFeePerGas: this.latestMaxPriorityFeePerGasGwei,
+        lastFastPriceGwei: this.lastFastPriceGwei,
       });
     }
   }
 
-  // Returns the current fast gas price in Wei, converted from the stored Gwei value.
+  // Returns the current fast maxFeePerGas and maxPriorityFeePerGas.
   getCurrentFastPrice() {
-    // Sometimes the multiplication by 1e9 introduces some error into the resulting number,
-    // so we'll conservatively ceil the result before returning. This output is usually passed into
-    // a web3 contract call so it MUST be an integer.
-    return Math.ceil(this.lastFastPriceGwei * 1e9);
+    // Sometimes the multiplication by 1e9 introduces some error into the resulting number, so we'll conservatively ceil
+    // the result before returning.This output is usually passed into a web3 contract call so it MUST be an integer.
+    if (this.type == "london") {
```

Comment:
this is the core idea of this PR: return an object with a different structure depending on the network type. then, when using this variable, use the spread operator (`...`) to expand this into the transaction object's settings.